### PR TITLE
Omnitab integration

### PR DESCRIPTION
**Pull request by jzbjyb**
# Adding OmniTab training and inference support to primeqa      

## What does this PR do?

```This PR adds scripts and instructions required to support training and inference of omnitab based table question answering model ```

### CODE REVIEW

File: notebooks/tableqa/README.md [hunk 1]
```diff
@@ -1 +1 @@
-This folder contains notebooks that demonstrates how to use [TAPAS](https://aclanthology.org/2020.acl-main.398.pdf) and [Tapex](https://openreview.net/pdf?id=O50443AsCP) from PrimeQA. Note that, PrimeQA supports train, eval, predict for both Tapas and Tapex. Pleae check the notebooks for more guidance on the usage.
+This folder contains notebooks that demonstrates how to use [TAPAS](https://aclanthology.org/2020.acl-main.398.pdf), [Tapex](https://openreview.net/pdf?id=O50443AsCP) and [OmniTab](https://aclanthology.org/2022.naacl-main.68.pdf) from PrimeQA. Note that, PrimeQA supports train, eval, predict for Tapas, Tapex and OmniTab. Pleae check the notebooks for more guidance on the usage.
```
Comment (riyazbhat) [inline notebooks/tableqa/README.md:1]: PrimeQA supports train, eval, `and` predict
Please add `and`

File: notebooks/tableqa/omnitab_inference.ipynb [hunk 2]
```diff
@@ -0,0 +1,302 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# OmniTab based Table Question Answering: Inference example\n",
+    "\n",
+    "In this notebook, we will show how to use our pretrained TableQA model based on OmniTab to answer questions over a table. The pre-trained model is available in huggingface and is trained on WikiTableQuestions dataset."
```
Comment (riyazbhat) [inline notebooks/tableqa/omnitab_inference.ipynb:9]: The pre-trained model is available `on`

Comment (riyazbhat) [reply notebooks/tableqa/omnitab_inference.ipynb:9]: and is trained on `the` WikiTableQuestions dataset

File: notebooks/tableqa/omnitab_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,342 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# OminTab based Table Question Answering: Training Example\n",
+    "In this notebook, we will show how to use our train a TableQA model based on tapex to answer questions over a table. The pre-trained model is available in huggingface and is trained on wiktablequestions dataset."
```
Comment (riyazbhat) [inline notebooks/tableqa/omnitab_training.ipynb:8]: is trained on `the` wiktablequestions dataset

Comment (riyazbhat) [reply notebooks/tableqa/omnitab_training.ipynb:8]: use hyphen `OminTab-based`

Comment (riyazbhat) [reply notebooks/tableqa/omnitab_training.ipynb:8]: is available `on` huggingface

File: primeqa/tableqa/tapex/README.md [hunk 4]
```diff
@@ -75,4 +75,56 @@ The trained model yields the following results on wikisql dev set:
   eval_steps_per_second    =      4.748
 ```
 
-TAPEX can also be used from PrimeQA built-in-class TapexReader to do train/eval/inference with minimal line of codes. See example [notebooks](https://github.com/primeqa/primeqa/tree/tapex_integration/notebooks/tableqa) for the same. 
+
+## Omnitab based TableQA models
+
+OmniTab is an omnivorous pretraining approach that consumes natural data to endow models with the ability to understand and align natural language with tables, and synthetic questions to train models to perform reasoning. More details can be found in the original paper [OmniTab: Pretraining with Natural and Synthetic Data for Few-shot Table-based Question Answering](https://arxiv.org/pdf/2207.03637.pdf).
+
+PrimeQA supports OmniTab based tableqa model training and inference over wikitablequestions dataset
```
Comment (riyazbhat) [inline primeqa/tableqa/tapex/README.md:83]: supports `OmniTab-based `
add hyphen

Comment (riyazbhat) [reply primeqa/tableqa/tapex/README.md:83]: over `the` wikitablequestions dataset
add `the`

File: primeqa/tableqa/tapex/README.md [hunk 5]
```diff
@@ -75,4 +75,56 @@ The trained model yields the following results on wikisql dev set:
...
+Before continuing below make sure you have PrimeQA [installed](https://primeqa.github.io/primeqa/installation.html).
+
+
+### Train/Eval using OmniTab based Table Question Answering model in PrimeQA on wikitablequestions dataset
```
Comment (riyazbhat) [inline primeqa/tableqa/tapex/README.md:87]: `OmniTab-based`
add hyphen

Comment (riyazbhat) [reply primeqa/tableqa/tapex/README.md:87]: PrimeQA on `the` wikitablequestions dataset
add `the`

File: primeqa/tableqa/tapex/README.md [hunk 6]
```diff
@@ -75,4 +75,56 @@ The trained model yields the following results on wikisql dev set:
...
+  --evaluation_strategy steps \
+  --predict_with_generate \
+  --num_beams 5 \
+  --weight_decay 1e-2 \
+  --label_smoothing_factor 0.1 \
+  --generation_max_length 128 \
+  --max_steps 20000 
+
+```
+The trained model yields the following results on wikitablequestions dev set:
```
Comment (riyazbhat) [inline primeqa/tableqa/tapex/README.md:117]: results on `the` wikitablequestions 
add `the`

**Review by riyazbhat — APPROVED** [review]
LGTM

File: notebooks/tableqa/omnitab_training.ipynb [hunk 7]
```diff
@@ -0,0 +1,342 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# OminTab based Table Question Answering: Training Example\n",
```
Comment (avisil) [inline notebooks/tableqa/omnitab_training.ipynb:7]: there's a typo: should be Omni `NOT` Omin.

File: notebooks/tableqa/omnitab_training.ipynb (hunk 3 above)
Comment (avisil) [inline notebooks/tableqa/omnitab_training.ipynb:8]: available via the HF Transformers library

**Review by avisil — APPROVED** [review]
LGTM - please correct the couple of minor typos which are there. Else great.

**Review by ma-sultan — APPROVED** [review]
LGTM. Left some comments.

**Pull request merged**